remsfal-services/remsfal-platform/src/main/java/de/remsfal/service/control/ContractorController.java

```diff
@@ -0,0 +1,198 @@
+package de.remsfal.service.control;
+
+import de.remsfal.core.model.ContractorModel;
+import de.remsfal.core.model.UserModel;
+import de.remsfal.service.entity.dao.ContractorRepository;
+import de.remsfal.service.entity.dao.ProjectRepository;
+import de.remsfal.service.entity.dto.ContractorEntity;
+import de.remsfal.service.entity.dto.ProjectEntity;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.NotFoundException;
+import org.jboss.logging.Logger;
+
+import java.util.List;
+
+/**
+ * Controller for contractor operations.
+ */
+@RequestScoped
+public class ContractorController {
+
+    @Inject
+    Logger logger;
+
+    @Inject
+    ProjectController projectController;
+
+    @Inject
+    ProjectRepository projectRepository;
+
+    @Inject
+    ContractorRepository contractorRepository;
+
+    /**
+     * Verify that the user has access to the project.
+     *
+     * @param user the user
+     * @param projectId the project ID
+     * @throws ForbiddenException if the user has no membership in the project
+     */
+    private void verifyProjectAccess(final UserModel user, final String projectId) {
+        try {
+            projectController.getProject(user, projectId);
+        } catch (NotFoundException e) {
+            throw new ForbiddenException("User has no membership in this project");
+        }
+    }
+
+    /**
+     * Verify that the user has access to the project and has privileged rights.
+     *
+     * @param user the user
+     * @param projectId the project ID
+     * @throws ForbiddenException if the user has no membership in the project or inadequate rights
+     */
+    private void verifyProjectAccessWithPrivilegedRights(final UserModel user, final String projectId) {
+        verifyProjectAccess(user, projectId);
+
+        if (!projectController.getProjectMemberRole(user, projectId).isPrivileged()) {
+            throw new ForbiddenException("Inadequate user rights");
+        }
+    }
+
+    /**
+     * Get contractors for a project.
+     *
+     * @param user the user
+     * @param projectId the project ID
+     * @param offset the offset
+     * @param limit the limit
+     * @return the list of contractors
+     */
+    public List<ContractorModel> getContractors(final UserModel user, final String projectId,
+                                               final Integer offset, final Integer limit) {
+        logger.infov("Retrieving contractors for project (id = {0})", projectId);
+        // Verify user has access to the project
+        verifyProjectAccess(user, projectId);
```

Comment:
This authorization check should not take place in the business logic, but rather in the boundary. And this already happens in the ContractorController with checkReadPermissions(projectId). So you could simply remove this check.